src/contractor/contractor.cpp

```diff
@@ -334,12 +335,17 @@ int Contractor::Run()
     }
 
     util::DeallocatingVector<QueryEdge> contracted_edge_list;
-    ContractGraph(max_edge_id,
-                  edge_based_edge_list,
-                  contracted_edge_list,
-                  std::move(node_weights),
-                  is_core_node,
-                  node_levels);
+    { // own scope to not keep the contractor around
+        GraphContractor graph_contractor(
```

Comment:
Not part of the PR scope: What about making the GraphContractor into a function? Having this `Constructor(), Run(), Get()` paradigmatic always seemed kinda weird to me here, since you actually can't call `Run()` more then once.